openroad_interface/macro_maker.py

```diff
@@ -0,0 +1,148 @@
+import math
+import copy
+import yaml
+
+from functions import *
+
+lef_tech_file = "OpenROAD/test/Nangate45/Nangate45_tech.lef" 
+spacing = None
+manufacturing_grid = None
+pin_size = 1000
+VDD = 1000
+VSS = 1000
+row_height = None
+spacing_constant = 3 
+die_coord_x2 = 100.13 * 1000
+die_coord_y2 = 100.8 * 1000
+
+######################################################################
+#######                                                        #######
+#######          all units are converted to nanometer          #######
+#######                x 1000 is the conversion                #######
+#######                                                        #######
+######################################################################
+
+# this script generates txt files that contain lef documentation for different generated macros
+
+def best(list):
```

Comment:
make general script that takes lef file and list of operators as inputs. use Arg parser.
chose appropriate key in tech_params based on lef choice.
get aspect ratio closest to Golden aspect ratio (set in vars).
let rowheights vary from 1 - 34. 
out put the macro text for the chosen aspect ratio. 
this is the last thing to do. 
(maybe we can do this in a separate PR).